src/test/java/api/loans/OverrideRenewByBarcodeTests.java

```diff
@@ -0,0 +1,530 @@
+      renewedLoan.getString("itemId"), is(smallAngryPlanet.getId().toString()));
+
+    assertThat("status should be open",
+      renewedLoan.getJsonObject("status").getString("name"), is("Open"));
+
+    assertThat("action should be renewed",
+      renewedLoan.getString("action"), is("Renewed through override"));
+
+    assertThat("'actionComment' field should contain comment specified for override",
+      renewedLoan.getString("actionComment"), is(OVERRIDE_COMMENT));
+
+    assertThat("renewal count should be incremented",
+      renewedLoan.getInteger("renewalCount"), is(1));
+
+    DateTime expectedDueDate = loanDueDate.plusWeeks(3).plusMonths(2);
+    assertThat("due date should be 1st of Feb 2019",
+      renewedLoan.getString("dueDate"),
+      isEquivalentTo(expectedDueDate));
+  }
+
+  @Test
+  public void canOverrideRenewalWhenLoanReachedRenewalLimitAndDueDateIsNotSpecified() throws
+    InterruptedException,
+    ExecutionException,
+    TimeoutException,
+    MalformedURLException {
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    LoanPolicyBuilder limitedRenewalsPolicy = new LoanPolicyBuilder()
+      .withName("Limited Renewals policy")
+      .rolling(Period.weeks(1))
+      .limitedRenewals(1);
+
+    UUID limitedRenewalsPolicyId = loanPolicyClient.create(limitedRenewalsPolicy).getId();
+
+    //Need to remember in order to delete after test
+    policiesToDelete.add(limitedRenewalsPolicyId);
+
+    useLoanPolicyAsFallback(limitedRenewalsPolicyId);
+
+    DateTime loanDate = DateTime.now();
+    loansFixture.checkOutByBarcode(smallAngryPlanet, jessica, loanDate).getJson();
+
+    loansFixture.renewLoan(smallAngryPlanet, jessica);
+
+    final JsonObject renewedLoan =
+      loansFixture.overrideRenewalByBarcode(smallAngryPlanet, jessica, OVERRIDE_COMMENT, null)
+        .getJson();
+
+    assertThat("user ID should match barcode",
+      renewedLoan.getString("userId"), is(jessica.getId().toString()));
+
+    assertThat("item ID should match barcode",
+      renewedLoan.getString("itemId"), is(smallAngryPlanet.getId().toString()));
+
+    assertThat("status should be open",
+      renewedLoan.getJsonObject("status").getString("name"), is("Open"));
+
+    assertThat("action should be renewed",
+      renewedLoan.getString("action"), is("Renewed through override"));
+
+    assertThat("'actionComment' field should contain comment specified for override",
+      renewedLoan.getString("actionComment"), is(OVERRIDE_COMMENT));
+
+    assertThat("renewal count should be incremented",
+      renewedLoan.getInteger("renewalCount"), is(2));
+
+    DateTime expectedDueDate = loanDate.plusWeeks(3);
+    assertThat("due date should be 3 weeks later",
+      renewedLoan.getString("dueDate"),
+      isEquivalentTo(expectedDueDate));
+  }
+
+  @Test
+  public void cannotOverrideRenewalWhenLoanDoesNotMatchAnyOfOverrideCases() throws
```

Comment:
I tried using the loan and policy in this example for a normal renewal (by replacing the override attempt locally) and that succeeded.

That suggests to me that this example demonstrates that a valid renewal cannot be overridden, rather than one that fails for a reason different to those expected.

Does that fit with your understanding?

Is this a compromise, as the implementation does not check for only the specific failure reasons, rather in some cases, any failure reason?

Reply:
Yes, this example demonstrates that valid renewal cannot be overriden.
Only renewals that fail for expected reasons (4 cases) can be overridden.
Does it make sense for you?

Reply:
> Is this a compromise, as the implementation does not check for only the specific failure reasons, rather in some cases, any failure reason?

Do you mean that place, when all expected override cases had failed and general error message is returned (something like 'loan does not match any override case')?
If so, I'm not sure how to implement it in different way.

Reply:
Yes, this example demonstrates that valid renewal cannot be overriden.
Only renewals that fail for expected reasons (4 cases) can be overridden.
Does it make sense for you?

Reply:
> Is this a compromise, as the implementation does not check for only the specific failure reasons, rather in some cases, any failure reason?
Do you mean due date calculation failure reasons? If so I probably gave the answer in the comment section above.

Reply:
Yes, I think you answered the question above, thank you.